src/protocol_edge.c

```diff
@@ -314,7 +356,19 @@ bool del_edge_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
 
 	/* Tell the rest about the deleted edge */
 
-	forward_request(mesh, c, request);
+
+	if(e->from->submesh && e->to->submesh && (e->from->submesh == e->to->submesh)) {
```

Comment:
This looks wrong. This should use the same checks as `add_edge_h()` does: if both sides of an edge are part of a submesh, they should not be in the same submesh, if they are then we should log an error and `return false`.

Reply:
I have changed this condition to work similar to add_edge_h.